code/circuit.py

```diff
@@ -0,0 +1,24 @@
+class circuit:
+
+    # a list of lines. The line should be in order
+    connection = []
```

Comment:
You can delete this line of code because Python is a little different than Java. In Python, we don't have to declare or initialize the `fields`* outside of the constructor. Instead, we just initialize them in the constructor (as you have done), and that's all we need to do.

* Things like `connection` are where an object stores its data. When an object-oriented programmer wants to talk about something like `connection`--but the programmer doesn't want to use terminology from a specific programming language--the programmer will often say "field" or "property" or "member".  When talking about something like `connection` In _Java_, people sometimes use the term "private variables" or  "private data members". That's because, in Java, we often use a private variable to implement a field. But I've avoided using the term "private variable" in my comments above because Python doesn't have private variables.